hare/broker.go

```diff
@@ -155,21 +112,22 @@ func (b *Broker) handleMessage(ctx context.Context, msg []byte) error {
 	logger.Debug("broker received hare message")
 
 	msgLayer := hareMsg.Layer
-	if !b.Synced(ctx, msgLayer) {
-		return errNotSynced
-	}
 
-	isEarly := false
-	if err = b.validateTiming(ctx, &hareMsg); err != nil {
-		if !errors.Is(err, errEarlyMsg) {
-			// not early, validation failed
-			logger.With().Debug("broker received a message to a consensus process that is not registered",
-				log.Err(err))
-			return err
+	out, exist := b.outbox[msgLayer.Uint32()]
+	if !exist {
+		logger.Debug("broker received a message to a consensus process that is not registered, latestLayer: %v, messageLayer: %v, messageType: %v", hareMsg.InnerMsg.Type)
+
+		// We consider messages for the next layer to be early and we process
+		// them, all other messages we ignore. We don't impose the outbox
+		// capacity at this point since this is an externally initiated event
+		// and at most it can cause us to exceed our capacity by one slot.
+		if msgLayer == b.latestLayer.Add(1) {
+			out = make(chan any, inboxCapacity)
+			b.outbox[msgLayer.Uint32()] = out
+		} else {
+			// This error is never inspected except to determine if it is not nil.
+			return errors.New("")
```

Comment:
i don't think this is a good practice. upstream callers can change and be surprised by the lack of msg in this error.
it should at least contain the layer/round of the message to provide context.

Reply:
Agreed :+1: